Add DB insertion functions

erikd opened the pull request:
This includes tests that are almost certain to fail with DB connections issues.

erikd committed:
core: Add DB insertion functions for all DB records

erikd committed:
core: Allow for silent migrations

erikd committed:
core: Simple initial database test

erikd committed:
stack.yaml: Use specific versions of persistent and hedgehog

dcoutts commented on cardano-explorer-core/test/test-db.hs:
```diff
@@ -0,0 +1,19 @@
+
+testMigration :: IO ()
+testMigration =
+  runMigrations True (PGPassFile "../config/pgpass") (MigrationDir "../schema") (LogFileDir "..")
```
Is `config/pgpass` something that should exist in the repo, or does one have to make one to run the test?

erikd replied:
Yeah, need to figure that out.

erikd replied:
It has been figured out. Subsequent commits on this branch add that file (without any credentials).

dcoutts reviewed (approved):
LGTM.

erikd committed:
core: Add a testInsert function

erikd committed:
core: Allow PGPassFile to use wildcard user

If the username in the PGPassFile is set to "*", use
`getEffectiveUserName` to get the user name.

erikd committed:
Add config/pgpass

No credentials here. Connects to a locally running PostgreSQL instance
using a unix domain socket and specifies the database name ("cexplorer").

erikd committed:
Nix voodoo

cleverca22 committed:
fix isIohkSkeleton so tests appear in release.nix

erikd committed:
core: Use PGPASSFILE env var instead of FilePath

Nix builds and tests each cabal project in a sandbox, so using a
file path outside that sandbox doesn't work. Instead, use the
PostgreSQL standard 'PGPASSFILE' environment variable.
